packages/grafana-data/src/types/datasource.ts

```diff
@@ -351,6 +351,19 @@ abstract class DataSourceApi<
    * @alpha -- experimental
    */
   getDefaultQuery?(app: CoreApp): Partial<TQuery>;
+
+  /*
+   * Optionally, use this method to track information about queries.
+   * Only executed once, during the dashboard initialization.
+   * @alpha -- experimental
+   */
+  trackQueries?(
+    queries: TQuery[],
+    dashboardId?: string,
+    orgId?: number,
+    userId?: number,
+    grafanaVersion?: string
```

Comment:
Is it possible to also include the plugin version of the data source?

Reply:
🤔 I did try but it does not seem to be accessible from the `DatasourceSrv` (neither in the DS nor in the `DatasourceInstanceSettings`) since `getVersion` is not implemented for azure.
I also tried using `this.azureMonitorDatasource.apiVersion` but that returns a default string `2018-01-01` not very useful.

It seems to be available in `CatalogPlugin` (that's what we show in the plugin's page), but I'm not sure if that's something I could (or want to) have access to here.

Any ideas on how to get that version ? Maybe @andresmgot or @vickyyyyyyy ?

Reply:
Note that this is only the information passed from Grafana to the plugin, then, the plugin will decide whatever it wants to track, in this case the information [here](https://github.com/grafana/grafana/pull/52052/files#diff-f285db3fb95d59ad6dc80121d21524a0a5c178d4c71f8a6c6f964b058372a814R218).

For core plugins, it doesn't make sense to additionally add the plugin version since it will always be the same. The Grafana version is the one that should be used.

For external plugins (e.g. ADX), there is probably a function to get the version (not the `CatalogPlugin` since that is not available externally) but that is not something that needs to be covered in this PR.

Reply:
Thanks, for external plugins, will there be another PR? Or you are saying that each plugin should do its own logging with this info?

Reply:
yes, by default plugins won't do any tracking. If they want, they would need to implement the `trackQueries` method and include there the information relevant to that plugin.